test/side-entrance/README.md

```diff
@@ -0,0 +1,380 @@
+        canWithdraw = true;
+    }
+    
+    // used for withdrawing ether balance in the pool
+    function withdraw() external {
+        if (canWithdraw) pool.withdraw();
+    }
+
+    // call the flashloan function of the pool, with a fuzzed amount
+    function flashLoan(uint amount) external {
+        pool.flashLoan(amount);
+    }
+
+    receive() external payable {}
+}
+```
+In my opinion, using such a **Handler** is not enough to say that the fuzzing is really capable of finding this kind of bugs.
+
+Another solution is made by the **Crytic team** and can be found [at this link](https://github.com/crytic/building-secure-contracts/blob/master/program-analysis/echidna/exercises/exercise7/solution.sol). Here the situation is much better: the solution is abstract enough and gives space for **Echidna** itself to find a bug. Besides, it only took a few seconds to solve it.
+
+Let's compare how **Echidna** and **Halmos** deal with the problem of "taking into account that any function can be executed inside **execute()**".
+
+Echidna:
+```solidity
+...
+function setEnableWithdraw(bool _enabled) public {
+    enableWithdraw = _enabled;
+}
+
+function setEnableDeposit(bool _enabled, uint256 _amount) public {
+    enableDeposit = _enabled;
+    depositAmount = _amount;
+}
+
+function execute() external payable override {
+    if (enableWithdraw) {
+        pool.withdraw();
+    }
+    if (enableDeposit) {
+        pool.deposit{value: depositAmount}();
+    }
+}
+...
+```
+We explicitly indicate which functions can be called and with which parameters. Obviously, if there was a larger setup - this code would become much more "bloated". We already saw something like this in ["Truster"](https://github.com/igorganich/damn-vulnerable-defi-halmos/tree/master/test/truster#echidna).
+
+Now Halmos:
+```solidity
+function execute () external payable {
+    ...
+    uint256 ETH_val = svm.createUint256("ETH_val_execute");
+    address target = svm.createAddress("target_execute");
+    bytes memory data;
+    (target, data) = glob.get_concrete_from_symbolic(target);
+    target.call{value: ETH_val}(data);
+    ...
+}
+```
+It is easy to see that the Halmos-based code provides a better abstraction for such cases and does a better job of expanding the setup.
+## Conclusions
+1. Using already accumulated techniques and principles, we solved the next Damn Vulnerable Defi challenge with Halmos quite easily. Every step was obvious and self-explanatory.
+2. Adapting the test to a specific contract is a good idea. For example, in this challenge we adapted to use native **ETH**.
+3. We confirm again the conclusions we made earlier: in the case of a small setup, fuzzing really seems to be a very effective tool, even if it is necessary to use some transaction abstraction. However, fuzzing engines do not have convenient abstraction mechanisms, so if target contracts are tied to some logic of abstract calls, Halmos looks much more convenient and powerful.
```

Comment:
nice comparison, and I agree with your conclusions! 

maybe a small observation: the new attacker developed here is more general/powerful than the one developed in previous scenarios, since it can receive and send ether. It may be worth noting that this more general attacker should also be able to find the solution to the previous challenges, right?